src/core/src/sim/mc/moves/mod.rs

```diff
@@ -65,6 +70,76 @@ fn select_molecule(system: &System, moltype: Option<u64>, rng: &mut Box<Rng>) ->
     }
 }
 
+/// This struct keeps track of the number of times a move was called
+/// and how often it was accepted. It also stores the current maximum displacement,
+/// i.e. the magnitude of a particle's translation/rotation or a volume's change in size.
+/// 
+/// We can modify (`update`) the maximum displacement to achieve a target acceptance rate 
+/// of a move based on the running acceptance.
+pub struct Displacement {
+    /// Current maximum displacement
+    delta: f64,
+    /// Current sample range
+    pub delta_range: Range<f64>,
+    /// Number of move calls
+    pub ncalls: u64, 
+    /// Total: number of accepted moves
+    pub naccepted: u64,
+    /// Total: number of attempted moves
+    pub nattempted: u64,
+    /// Target acceptance for the move
+    target_acceptance: f64,
+    /// Call updates with this frequency
+    update_frequency: u64,
+}
+
+impl Displacement {
+    pub fn new(delta: f64, target_acceptance: f64, update_frequency: u64) -> Displacement {
+        Displacement {
+            delta: delta,
+            delta_range: Range::new(-delta, delta),
+            ncalls: 0,
+            naccepted: 0,
+            nattempted: 0,
+            target_acceptance: target_acceptance,
+            update_frequency: update_frequency,
+        }
+    }
+    
+    /// Change `update_frequency` to zero, which will prevent further updates.
+    pub fn disable_updates(&mut self) { self.update_frequency = 0; }
+    
+    /// Update the current maximum displacement every `update_frequency`th time a move is called
+    /// to achieve a given acceptance during the course of a simulation.
+    /// Note that by changing the displacement during the simulation run, you no longer produce a 
+    /// Markov chain. For production runs (or after some MC steps) you should switch the `update_frequency`
+    /// to zero (`disable_updates`) to avoid further updates.
+    pub fn update(&mut self) {
+        // Don't update if `update_frequency` is zero or we are not within `update_frequency`
+        // Also, check if attempts are zero (first call or reset) which will lead to division by zero.
+        if self.update_frequency == 0 || self.ncalls % self.update_frequency != 0 || self.nattempted == 0 {
+            return
+        } 
+        // get current acceptance
+        let current_acceptance = self.naccepted as f64 / self.nattempted as f64;
+        // limit changes:
+        // if quotient exceeds a min/max value of 50%, limit changes to 50%.
+        let quotient = current_acceptance / self.target_acceptance;
+        match quotient {
+            _ if quotient > 1.5 => self.delta *= 1.5,
+            _ if quotient < 0.5 => self.delta *= 0.5,
```

Comment:
50% seems to be a very big change. I have seen 10% in other software, do you have a specific reason to pick 50% ?